In [25]:
import pandas as pd
import numpy as np

In [26]:
df = pd.read_csv('mobile_events_2020.csv',sep = ';')

In [27]:
df.head(10)

anonymous_id business_zoned context_app_version  \
0  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
1  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
2  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
3  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
4  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
5  5592E78E-CE87-4258-B8D7-DD834A687A1E        Hamburg              3.1.13   
6  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
7  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
8  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   
9  9DDC5BCD-0477-4E83-A32A-51013DA0A4E7        Hamburg              3.1.13   

                      context_device_id context_os_name  \
0  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
1  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
2  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
3  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
4  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
5  11D0CB16-DA38-4565-9C3B-066E7877856C             iOS   
6  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
7  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
8  7A718FF6-584F-40E2-939D-224231295B1E             iOS   
9  7A718FF6-584F-40E2-939D-224231295B1E             iOS   

                                          event_name          received_at  \
0                    Out-Of-Business Zone View Shown  2020-09-01 23:59:44   
1                                 Application Opened  2020-09-01 23:59:41   
2                             Ride Done - Successful  2020-09-01 23:59:41   
3                                 Application Opened  2020-09-01 23:59:36   
4                           Ride Done - Unsuccessful  2020-09-01 23:59:36   
5                                 Application Opened  2020-09-01 23:59:36   
6                                 Application Opened  2020-09-01 23:59:14   
7  Reduced Speed Snackbar Shown - Out Of Business...  2020-09-01 23:59:14   
8                    Out-Of-Business Zone View Shown  2020-09-01 23:59:14   
9                                 Application Opened  2020-09-01 23:58:58   

            created_at  
0  2020-09-01 23:59:18  
1  2020-09-01 23:59:35  
2  2020-09-01 23:59:37  
3  2020-09-01 23:37:46  
4  2020-09-01 23:37:52  
5  2020-09-01 23:57:45  
6  2020-09-01 23:58:59  
7  2020-09-01 23:59:03  
8  2020-09-01 23:59:03  
9  2020-09-01 23:58:58

In [28]:
df.tail(10)

anonymous_id business_zoned  \
102412  06722623-7FBB-4368-90A5-9770433A41EB        Hamburg   
102413  7C6780F1-B0F4-4525-B008-09591C5DD214        Hamburg   
102414  7C6780F1-B0F4-4525-B008-09591C5DD214        Hamburg   
102415  7C6780F1-B0F4-4525-B008-09591C5DD214        Hamburg   
102416  7C6780F1-B0F4-4525-B008-09591C5DD214        Hamburg   
102417  7C6780F1-B0F4-4525-B008-09591C5DD214        Hamburg   
102418  E1F50677-B43A-47DE-88A9-BD5A552B7D9E        Hamburg   
102419  3569EA33-48C7-4E02-8B22-9E613C40481C        Hamburg   
102420  46f981e2-d66d-43da-922d-38f4c14d02dd        Hamburg   
102421  7e842b89-f8e9-492d-978d-8dda318eca15        Hamburg   

       context_app_version                     context_device_id  \
102412              3.1.13  FAE8C3BE-C655-4AB9-AB3F-D80380D8D6FE   
102413              3.1.13  0B5D7461-69F7-436A-8960-B806EEDCA5B4   
102414              3.1.13  0B5D7461-69F7-436A-8960-B806EEDCA5B4   
102415              3.1.13  0B5D7461-69F7-436A-8960-B806EEDCA5B4   
102416              3.1.13  0B5D7461-69F7-436A-8960-B806EEDCA5B4   
102417              3.1.13  0B5D7461-69F7-436A-8960-B806EEDCA5B4   
102418              3.1.13  0D0FD67D-303A-4CD6-8C8B-8F033204ACED   
102419              3.1.13  0CBB29F5-548F-4A88-B4A3-0284FC03F1FD   
102420               3.1.9  46f981e2-d66d-43da-922d-38f4c14d02dd   
102421               3.1.9  7e842b89-f8e9-492d-978d-8dda318eca15   

       context_os_name                                        event_name  \
102412             iOS                                Application Opened   
102413             iOS                                Application Opened   
102414             iOS                                 Qr Scanner Tapped   
102415             iOS                    Qr Scanner - Scanned Succeeded   
102416             iOS  Qr Scanner - Scan Failed - Scooter Not Available   
102417             iOS                         Qr Scanner - Enter Number   
102418             iOS                                Application Opened   
102419             iOS                               Qr Scanner - Cancel   
102420         Android                           Scooter Selected - Auto   
102421         Android                           Scooter Selected - Auto   

                received_at           created_at  
102412  2020-09-01 00:00:07  2020-09-01 00:00:03  
102413  2020-09-01 00:00:05  2020-08-31 23:59:30  
102414  2020-09-01 00:00:05  2020-08-31 23:59:36  
102415  2020-09-01 00:00:05  2020-08-31 23:59:50  
102416  2020-09-01 00:00:05  2020-08-31 23:59:50  
102417  2020-09-01 00:00:05  2020-08-31 23:59:50  
102418  2020-09-01 00:00:04  2020-08-31 23:59:59  
102419  2020-09-01 00:00:03  2020-08-31 23:59:55  
102420  2020-09-01 00:00:03  2020-08-31 23:59:41  
102421  2020-09-01 00:00:02  2020-08-31 23:59:58

In [29]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102422 entries, 0 to 102421
Data columns (total 8 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   anonymous_id         102422 non-null  object
 1   business_zoned       102422 non-null  object
 2   context_app_version  102422 non-null  object
 3   context_device_id    102422 non-null  object
 4   context_os_name      102422 non-null  object
 5   event_name           102422 non-null  object
 6   received_at          102422 non-null  object
 7   created_at           102422 non-null  object
dtypes: object(8)
memory usage: 6.3+ MB


In [30]:
string_search = 'Scooter Selected - Auto'

percentage_with_intention = (df['event_name'].isin([string_search]).sum() / df.shape[0]) * 100

In [31]:
percentage_with_intention

8.528441155220557

8.53% of sessions with an intention to book a scooter

2. What is the % of sessions with a successful booking? 3. What determines, according to you, an intention to book?
4. What are factors, according to you, determining a successful booking? 
5. Do you have any hypotheses after exploring this data? If yes, please add them.

# Question #2

In [34]:
string_search = 'Qr Scanner - Scanned Succeeded'

percentage_with_intention = (df['event_name'].isin([string_search]).sum() / df.shape[0]) * 100

In [35]:
percentage_with_intention

4.810489933803284

# Question #3

# Question #4

In [ ]:
Yes, Hypothsis testing can be done, We can use 

1.Hypothesis: Can we predict the "event_name" based on other variables like "business_zoned" or "context_app_version"?
    Test: Logistic regression for categorical outcomes or linear regression for continuous outcomes.
        
2. Proportion Test:

Hypothesis: Is there a significant difference in the proportion of "Scooter Selected - Auto" events between iOS and Android users?
Test: Chi-squared test for proportions.
    
3.Mean Test:

Hypothesis: Is there a significant difference in the mean "received_at" time between "Application Opened" events and "Qr Scanner Tapped" events?
Test: Two-sample t-test or ANOVA (Analysis of Variance) if comparing multiple event types.